src/main/java/event/EventProcessor.java

```diff
@@ -0,0 +1,17 @@
+package event;
+
+import action.Action;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+class EventProcessor {
```

Comment:
do not create separate class for just one operation. This makes reader very hard to navigate in your code. Reader sees EventDetector, EventProcessor, EventManager ... and becomes scared to understand all of them